### ISSUE TITLE
Introduce priority write/read options

### ISSUE THREAD

**Issue by jburse** [jburse/dogelog-moon#75]
The priority write option might be useful for answer substitution
display, since the writing of `Variable = Term` cannot be done
by writeq/1 directly, since it wouldn't display the Variable name

quoted. Unless we have some `'$VAR'/1` support and the 
convention that this compound can also take an atom. But even
for Jekejeke Prolog we didn't approach it that way, and

rather used fleshing out the `Variable`, the `=` and the `Term `with
different write calls. The Term must then be written with a 
priority different from the default priority 1200.

See also:

**Provide Dogelog query compilation #25**
[https://\github.com/jburse/dogelog-moon/issues/25](https://github.com/jburse/dogelog-moon/issues/25)

**Comment by jburse** [comment]
We have recently implemented variable quoting through
back quotes. We can reach into these routines to display
the variable name. We did something similar in

listing/[0,1] already, to display the terminating period.
The result is that Dogelog runtime does itself not boots-
trap from the ISO core standard on a higher level,

there is some low level access to the implementation
of write_term/[2,3] not covered by the ISO core standard.

**Comment by jburse** [comment]
Interestingly many Prolog systems only implement a priority/1
write option, but not a priority/1 read option.  One of the
few Prolog systems that implement both are SWI-Prolog

and Jekejeke Prolog, and now Dogelog runtime. Neither
the write option nor the read option is found in the ISO
core standard. We have currently only use for the priority

write option in the query answers, but we also provided the
read option for symmery. Its also the first read option, which
is not a read result. So we implemented read option decoding

for priority and could extend it later if further options are demanded.
